app/src/main/java/org/p2p/wallet/newsend/NewSendPresenter.kt

```diff
@@ -1,5 +1,45 @@
 package org.p2p.wallet.newsend
 
+import kotlinx.coroutines.launch
+import org.p2p.wallet.common.analytics.interactor.ScreensAnalyticsInteractor
 import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.home.analytics.BrowseAnalytics
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.user.interactor.UserInteractor
+import kotlin.properties.Delegates
 
-class NewSendPresenter() : BasePresenter<NewSendContract.View>(), NewSendContract.Presenter
+class NewSendPresenter(
+    private val userInteractor: UserInteractor,
+    private val browseAnalytics: BrowseAnalytics,
+    private val analyticsInteractor: ScreensAnalyticsInteractor,
+) : BasePresenter<NewSendContract.View>(), NewSendContract.Presenter {
+
+    private var token: Token.Active? by Delegates.observable(null) { _, _, newValue ->
+        if (newValue != null) view?.showSourceToken(newValue)
+    }
+
+    init {
+        launch {
+            token = userInteractor.getUserTokens().first()
```

Comment:
we can see this screen only if user has tokens?

Reply:
we just preselect SOL for this screen

Reply:
Are you sure it's SOL? 
I remember we talked about stable coins here. 
Please talk to task owner